diff -ruN .gitignore .gitignore
--- .gitignore	1970-01-01 01:00:00.000000000 +0100
+++ .gitignore	2022-08-16 18:19:39.564806091 +0200
@@ -0,0 +1,2 @@
+*
+!.gitignore
diff -ruN Token.sol Token.sol
--- Token.sol	2022-08-05 10:29:36.363758759 +0200
+++ Token.sol	2022-08-16 18:21:30.233687410 +0200
@@ -46,4 +46,29 @@
   function burn(uint256 amount) external {
     _burn(msg.sender, amount);
   }
+
+  // HARNESS: add underlying functions to the public interface to be able to compare them with the actual functions
+
+  function underlyingTransfer(address to, uint256 amount)
+    public
+    returns (bool)
+  {
+    return super.transfer(to, amount);
+  }
+
+  function underlyingTransferFrom(
+    address from,
+    address to,
+    uint256 amount
+  ) public returns (bool) {
+    return super.transferFrom(from, to, amount);
+  }
+
+  function underlyingMint(address to, uint256 amount) public {
+    _mint(to, amount);
+  }
+
+  function underlyingBurn(uint256 amount) public {
+    _burn(msg.sender, amount);
+  }
 }
